lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -236,26 +254,38 @@ def _on_start(self, event: StartEvent):
 
             return
 
-        if not self._is_tls_fully_configured():
-            self.status.set(BlockedStatus(TLSNotFullyConfigured))
+        # apply the directives computed and emitted by the peer cluster manager
+        if not self._apply_peer_cm_directives_and_check_if_can_start():
             event.defer()
             return
 
-        self.status.clear(TLSNotFullyConfigured)
-
-        # apply the directives computed and emitted by the peer cluster manager
-        if not self._apply_peer_cm_directives_and_start():
+        if not self.is_admin_user_configured() or not self.is_tls_fully_configured():
+            if not self.model.get_relation("certificates"):
+                status = BlockedStatus(TLSRelationMissing)
+            else:
+                status = MaintenanceStatus(
+                    TLSNotFullyConfigured
+                    if self.is_admin_user_configured()
+                    else AdminUserNotConfigured
+                )
+            self.status.set(status)
             event.defer()
             return
 
+        self.status.clear(AdminUserNotConfigured)
+        self.status.clear(TLSNotFullyConfigured)
+        self.status.clear(TLSRelationMissing)
+
+        self.peers_data.put(Scope.UNIT, "tls_configured", True)
```

Comment:
`tls_configured` is a flag on the unit databag which indicates whether TLS has been configured or not in a unit. 
The peer-cluster-relation however uses an aggregate of this flag fetched from all units, through `is_tls_full_configured_in_cluster()` - to get an overview of the TLS state in the entire cluster.

(Other than that we pass the CA in the relation)

Reply:
Yes.
The `tls` flag on the relation is to indicate whether the service is configured to use TLS.
So the same purpose as what `tls_configured` expresses on the node level.

That's why I'm wondering if we may wanna apply the same terminology...?

Reply:
Hmm, funny enough, I just checked [Charm Relation Interfaces](https://github.com/canonical/charm-relation-interface), and the `tls` flag doesn't seem required. (Makes sense as it's redundant with the presence of the `tls-ca` (secret) field.)

[All interfaces inherit the corresp function](https://github.com/canonical/data-platform-libs/blob/main/lib/charms/data_platform_libs/v0/data_interfaces.py#L1288), but  it may be really used anymore...?

Reply:
Thanks for the suggestion! i believe I need to take a closer look at the work you did in the data_interfaces to see everything we can benefit from! I suggest to do it in another PR